--- a/4.4/sensor-src/t41/gc5603.c	2026-02-05 20:59:18.215517908 +0000
+++ b/4.4/sensor-src/t41/gc5603.c	2026-02-05 20:59:12.141630784 +0000
@@ -1027,24 +1027,21 @@
 	}
 
 	switch(info->mclk) {
-		case TISP_SENSOR_MCLK0:
-			sclka = private_devm_clk_get(&client->dev, "mux_cim0");
-			sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim0");
-			set_sensor_mclk_function(0);
-			break;
-		case TISP_SENSOR_MCLK1:
-			sclka = private_devm_clk_get(&client->dev, "mux_cim1");
-			sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim1");
-			set_sensor_mclk_function(1);
-			break;
-		case TISP_SENSOR_MCLK2:
-			sclka = private_devm_clk_get(&client->dev, "mux_cim2");
-			sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim2");
-			set_sensor_mclk_function(2);
-			break;
-		default:
-			ISP_ERROR("Have no this MCLK Source!!!\n");
-	}
+        case TISP_SENSOR_MCLK0:
+            sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim");
+            set_sensor_mclk_function(0);
+            break;
+        case TISP_SENSOR_MCLK1:
+            sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim");
+            set_sensor_mclk_function(1);
+            break;
+        case TISP_SENSOR_MCLK2:
+            sensor->mclk = private_devm_clk_get(sensor->dev, "div_cim");
+            set_sensor_mclk_function(2);
+            break;
+        default:
+            ISP_ERROR("Have no this MCLK Source!!!\n");
+    }
 
 	if (IS_ERR(sensor->mclk)) {
 		ISP_ERROR("Cannot get sensor input clock cgu_cim\n");
