--- a/src/imp_hal.cpp	2026-03-01 20:35:30.287634364 +0000
+++ b/src/imp_hal.cpp	2026-03-01 20:35:18.627198486 +0000
@@ -765,7 +765,7 @@
   IMPISPAEExprInfo info{};
   int ret = IMP_ISP_Tuning_GetAeExprInfo(IMPVI, &info);
   if (ret == 0) {
-    out_gain = static_cast<int>(info.TotalGainDb);
+    out_gain = static_cast<int>(info.TotalGainDb) / 256;
   }
   return ret;
 #elif defined(PLATFORM_T10) || defined(PLATFORM_T20) || defined(PLATFORM_T21) || defined(PLATFORM_T23) || \
